src/Voron/Data/CompactTrees/PersistentDictionary.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+using Sparrow;
+using Sparrow.Server.Compression;
+using Voron.Exceptions;
+using Voron.Global;
+using Voron.Impl;
+using static Sparrow.Hashing;
+
+namespace Voron.Data.CompactTrees
+{
+    [StructLayout(LayoutKind.Explicit, Pack = 1)]
+    public unsafe struct PersistentDictionaryHeader
+    {
+        public const int SizeOf = 32;
+
+        [FieldOffset(0)]        
+        public ulong TableHash;
+        [FieldOffset(8)]
+        public long CurrentId;
+        [FieldOffset(16)]
+        public long PreviousId;
+
+        public override string ToString()
+        {
+            return $"{nameof(TableHash)}: {TableHash}, {nameof(CurrentId)}: {CurrentId}, {nameof(PreviousId)}: {PreviousId}";
+        }
+    }
+
+    public unsafe partial class PersistentDictionary 
+    {
+        private readonly Page _page;
+        
+        public const int UsableDictionarySize = NumberOfPagesForDictionary * Constants.Storage.PageSize - PageHeader.SizeOf - PersistentDictionaryHeader.SizeOf;
+
+        public long PageNumber => _page.PageNumber;
+        
+        private readonly HopeEncoder<Encoder3Gram<NativeMemoryEncoderState>> _encoder;
+        private byte[] _tempBuffer;
+
+        public static long CreateDefault(LowLevelTransaction llt)
+        {
+            var p = llt.AllocatePage(NumberOfPagesForDictionary);
+            p.Flags = PageFlags.Overflow;
+            p.OverflowSize = NumberOfPagesForDictionary * Constants.Storage.PageSize;
+
+            PersistentDictionaryHeader* header = (PersistentDictionaryHeader*)p.DataPointer;
+            header->CurrentId = p.PageNumber;
+            header->PreviousId = 0;
+
+            // We retrieve the embeeded file from the assembly, copy and checksum the entire thing. 
+            var dictionary = new byte[UsableDictionarySize];
+            var embeddedFile = typeof(PersistentDictionary).Assembly.GetManifestResourceStream($"Voron.Data.CompactTrees.dictionary.bin");
+            embeddedFile.Read(dictionary);
```

Comment:
This is a recipe for errors. 
When we update the dictionary definition, we'll have different size.
No idea what that would have. Let's do something the `ReadAllBytes` there, instead

Reply:
How would you _tell_ that you didn't read enough?

Reply:
You encoding + decoding of standard sequences would fail. That was exactly what I had in mind.

Reply:
why would they fail, instead of getting bad results?
In short, fix it so there isn't some convulted manner to say "this is broken"

Reply:
It is simple, if it fails a simple 'compress-decompress' cycle then we made a mistake during the building process. That check is been done once per instantiation, now the question is: "Should be doing that before even starting RavenDB or on usage?" 

I lean toward starting.

Reply:
The _code_ isn't simple, it is fragile, which means a change will break in subtle ways.
Just read it proeprly